pinot-core/src/main/java/com/linkedin/pinot/core/realtime/converter/stats/RealtimeColumnStatistics.java

```diff
@@ -67,21 +68,64 @@ public RealtimeColumnStatistics(RealtimeColumnDataSource dataSource, int[] sorte
 
   @Override
   public Object getMinValue() {
+    if (_dictionaryReader == null) {
+      switch (_dataSource.getDataSourceMetadata().getDataType()) {
+        case LONG:
+          return Long.MIN_VALUE;
+        case FLOAT:
+          return Float.MIN_VALUE;
+        case DOUBLE:
+          return Double.MIN_VALUE;
+        case INT:
+        default:
+          return Integer.MIN_VALUE;
+      }
+    }
     return _dictionaryReader.getMinVal();
   }
 
   @Override
   public Object getMaxValue() {
+    if (_dictionaryReader == null) {
+      switch (_dataSource.getDataSourceMetadata().getDataType()) {
+        case LONG:
+          return Long.MAX_VALUE;
+        case FLOAT:
+          return Float.MAX_VALUE;
+        case DOUBLE:
+          return Double.MAX_VALUE;
+        case INT:
+        default:
+          return Integer.MAX_VALUE;
+      }
+    }
     return _dictionaryReader.getMaxVal();
   }
 
   @Override
   public Object getUniqueValuesSet() {
+    if (_dictionaryReader == null) {
+      switch (_dataSource.getDataSourceMetadata().getDataType()) {
+        case LONG:
+          return new long[]{Long.MAX_VALUE};
+        case FLOAT:
+          return new float[]{Float.MAX_VALUE};
+        case DOUBLE:
+          return new double[]{Double.MAX_VALUE};
+        case INT:
+          return new int[]{Integer.MAX_VALUE};
+        default:
+          return new String[]{""};
+      }
+    }
     return _dictionaryReader.getSortedValues();
   }
 
   @Override
   public int getCardinality() {
+    if (_dictionaryReader == null) {
```

Comment:
this can break something? Doesn't group by etc depend on this. A better default is the number of rows in the segment?